docker/Dockerfile.ubuntu

```diff
@@ -0,0 +1,92 @@
+# This file aimed for developers and optimized for quick re-builds
+# Shouild be used by running `make localdocker`
+FROM ffmpeg-base:nvidia as builder
+
+FROM golang:1-stretch as builder2
+ENV PKG_CONFIG_PATH /root/compiled/lib/pkgconfig
+WORKDIR /root
+RUN apt update \
+    && apt install -y \
+    git gcc g++ gnutls-dev 
+    # git gcc g++ gnutls-dev linux-headers
+COPY --from=builder /root/compiled /root/compiled/
+#COPY --from=builder /usr/local/cuda /usr/local/cuda
+
+ENV PKG_CONFIG_PATH /root/compiled/lib/pkgconfig
+#ENV CPATH /usr/local/cuda/include 
+#ENV LIBRARY_PATH /usr/local/cuda/lib64
+
+WORKDIR /go/src/github.com/livepeer/go-livepeer
+COPY . .
+RUN go get -u -v ./...
+
+#RUN go get github.com/golang/glog
+#RUN go get -u -v github.com/ericxtang/m3u8
+#RUN go get github.com/aws/aws-sdk-go/aws
+#RUN go get -u google.golang.org/grpc
+#RUN go get github.com/pkg/errors
+#RUN go get github.com/stretchr/testify/mock
+#RUN go get -u -v go.opencensus.io/stats
+#RUN go get -u -v go.opencensus.io/tag
+#RUN go get -u -v contrib.go.opencensus.io/exporter/prometheus
+
+#COPY vendor vendor
+# .dockerbuild.deps contains list of packages used by go-client
+# needed to build them before building our code so it will be cached
+# by docker and reused on next builds.
+# it was generated manually by running `go build -v` and capturing output
+# should be updated only if we remove some dependency from vendor dir
+#COPY .dockerbuild.deps .dockerbuild.deps
+
+# this will pre-build and cache files in /vendor dir
+#RUN go list -v -export $(cat .dockerbuild.deps)
+#WORKDIR /go/src/github.com/livepeer/go-livepeer/cmd/livepeer
+#RUN go build -v ./...
+
+# ipfs, monitor and pm dirs special cased as presumanbly being
+# less frequent changed
+# adding 'export` to other dirs probably won't add any improvements
+# to re-build speed
+COPY ipfs ipfs
+RUN go list -v -export github.com/livepeer/go-livepeer/ipfs
+COPY monitor monitor
+RUN go list -v -export github.com/livepeer/go-livepeer/monitor
+COPY pm pm
+RUN go list -v -export github.com/livepeer/go-livepeer/pm
+COPY VERSION VERSION
+
+
+COPY cmd cmd
+COPY common common
+COPY core core
+COPY server server
+COPY discovery discovery
+COPY drivers drivers
+COPY net net
+COPY eth eth
```

Comment:
What's the purpose of copying to and from the same folder here?

Reply:
The `COPY` command in docker copies a folder during the build to the image being created, so for example when we run 
```bash
sudo docker build --rm -t livepeerbinary:latest -f docker/Dockerfile.debian .
```

the `.` is our source folder, so `COPY net net` would copy the folder  `go-livepeer/net` to the image working directory.

Reply:
Ah I see. It makes sense that it's being copied to the docker image! 💯